### PR TITLE
Added ability to highlight comments in repl

### LINKED ISSUE #5169: Support syntax highlighting for comments
See [src/dotty/tools/dotc/printing/SyntaxHighlighting.scala](https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/printing/SyntaxHighlighting.scala)

### PR DESCRIPTION
**Pull request by veera83372**
Fixes #5169

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/parsing/Scanners.scala [hunk 1]
```diff
@@ -177,6 +177,9 @@ object Scanners {
     /** All doc comments kept by their end position in a `Map` */
     private[this] var docstringMap: SortedMap[Int, Comment] = SortedMap.empty
 
+    /* All comments positions will be stored in this */
+    var commentPositions:mutable.MutableList[Position] = mutable.MutableList()
```
Comment (allanrenucci) [inline compiler/src/dotty/tools/dotc/parsing/Scanners.scala]: I suggest using a `ListBuffer` instead:
```scala
private[this] val commentPosBuf = new mutable.ListBuffer[Int]

/** Return a list of all the comment positions */
def commentPositions: List[Position] = commentPosBuf.toList
```

File: compiler/src/dotty/tools/dotc/printing/SyntaxHighlighting.scala [hunk 2]
```diff
@@ -48,6 +49,11 @@ object SyntaxHighlighting {
           highlightRange(pos.start, pos.end, color)
       }
 
+      def highlightComments(commentsPos: mutable.MutableList[Position]): Unit = {
```
Comment (allanrenucci) [inline compiler/src/dotty/tools/dotc/printing/SyntaxHighlighting.scala]: This is used only once. I would inline at call site

Comment (veera83372) [reply compiler/src/dotty/tools/dotc/printing/SyntaxHighlighting.scala]: Do you mean to say use `@inline` or replace call invocation with logic?

Comment (allanrenucci) [reply compiler/src/dotty/tools/dotc/printing/SyntaxHighlighting.scala]: replace call invocation with logic

File: compiler/src/dotty/tools/dotc/printing/SyntaxHighlighting.scala [hunk 3]
```diff
@@ -11,6 +11,7 @@ import dotty.tools.dotc.util.Positions.Position
 import dotty.tools.dotc.util.SourceFile
 
 import java.util.Arrays
+import scala.collection.mutable
```
Comment (allanrenucci) [inline compiler/src/dotty/tools/dotc/printing/SyntaxHighlighting.scala]: unused

File: compiler/src/dotty/tools/dotc/printing/SyntaxHighlighting.scala [hunk 4]
```diff
@@ -81,6 +82,10 @@ object SyntaxHighlighting {
         }
       }
 
+      scanner.commentPositions.foreach {
+        highlightPosition(_,CommentColor)
+      }
```
Comment (allanrenucci) [inline compiler/src/dotty/tools/dotc/printing/SyntaxHighlighting.scala]: nitpick: we usually don't use the underscore syntax with curly braces. I suggest writing
```scala
for (pos <- scanner.commentPositions)
  highlightPosition(pos, CommentColor)
```
or
```scala
scanner.commentPositions.foreach { pos =>
  highlightPosition(pos, CommentColor)
}
```
or
```scala
scanner.commentPositions.foreach(highlightPosition(_, CommentColor))
```

I personally prefer the first one

File: compiler/src/dotty/tools/dotc/printing/SyntaxHighlighting.scala (hunk 3 above)
Comment (veera83372) [reply compiler/src/dotty/tools/dotc/printing/SyntaxHighlighting.scala]: Removed in final commit

File: compiler/src/dotty/tools/dotc/printing/SyntaxHighlighting.scala (hunk 4 above)
Comment (veera83372) [reply compiler/src/dotty/tools/dotc/printing/SyntaxHighlighting.scala]: I have selected the first approach only

**Pull request merged**